main.js

```diff
@@ -0,0 +1,94 @@
+$(".post-button").click(function() {
+  var userName = $(".input-username").val();
+  var comment = $(".input-comment").val();
+
+  //handlebars
+  var newPost = {
+    userName: userName, comment: comment
+  }
+  var source = $("#post-template").html();
+  var template = Handlebars.compile(source);
+  var newHTML = template(newPost);
+  $('.comment-section').append(newHTML);
+  //!handlebars
+
+  //clear texboxes
+  $(".input-username").val('');
+  $(".input-comment").val('');
+
+  $('.remove-post').click(function () {
+    $(this).closest('.row').remove();
+  });
+
+
+  $('.edit-post').click(function(){
+    //save current post.
+    var oldPost = $($(this)[0]).siblings(".comment")[0].textContent;
+    var changes = {
+      oldPost: oldPost
+    }
+    var editSource = $("#edit-post-template").html();
+    var editTemplate = Handlebars.compile(editSource);
+    var editComment = editTemplate(changes);
+    // delete original post
+    $($(this)[0]).siblings(".comment").remove();
+    // add text box loaded with original post, add save button.
+    $(this).closest('div').append(editComment);
+
+
+    $('.save-changes').click(function(){
+      // save new input.
+      var updatedComment = $($(this)[0].parentNode).find(".input-comment").val();
```

Comment:
Why are all these variables named with `comment` when this is about editing the post?

Reply:
I see how that can be confusing. I used the .comment class in CSS to format any kind of comment/post. I should have added a separate class for the selector so it would have been more straight forward.